tools/ou3_p5_45deg_sample1_entry_ha.py

```
@@ -0,0 +1,127 @@
+  if d.get(k) is not True:f.append(k)
+ for k in ('source_replay_used','filter_changed','sample1_measurement_prefix_evaluated_here','returned_to_30deg_P4_sector_here','P4_COMPLETE_WORD_DISSIPATION_ESTABLISHED_HERE','P5_FINITE_CAPTURE_TO_P4_ESTABLISHED_HERE'):
+  if d.get(k) is not False:f.append(k)
+ if d.get('H_dimension')!=18 or d.get('A_dimension')!=21:f.append('H/A dimensions')
+ if d.get('sample1_entry_inside_q8') is not True:f.append('sample1 q8')
+ for mode,dim in (('H',18),('A',21)):
+  m=d.get('modes',{}).get(mode,{})
+  if m.get('dimension')!=dim or m.get('complete') is not True or m.get('evaluated')!=m.get('expected') or m.get('first_failure') is not None:f.append(f'{mode} incomplete')
+ return list(dict.fromkeys(f))
+
+def main():
+ a=argparse.ArgumentParser(); a.add_argument('--domain',type=Path,default=DEFAULT_DOMAIN); a.add_argument('--source-pieces',type=int,default=2); a.add_argument('--output',type=Path,required=True); x=a.parse_args(); d=build(x.domain,x.source_pieces); vf=validate(d); d['validation_pass']=not vf; d['validation_failures']=vf; x.output.write_text(json.dumps(d,indent=2,sort_keys=True)); print(json.dumps({'status':d['P5_45DEG_SAMPLE1_ENTRY_HA_CERTIFICATE'],'q1':d['sample1_pre_measurement_q_upper'],'H':{k:d['modes']['H'][k] for k in ('complete','max_state','first_failure')},'A':{k:d['modes']['A'][k] for k in ('complete','max_state','first_failure')},'next':d['next_obligation'],'validation_failures':vf},indent=2,sort_keys=True)); return 0 if not vf else 2
```

Ruff (0.16.3)

[error] Multiple statements on one line (semicolon)

(E702)